### ISSUE TITLE
Ability to reflect classes not loaded via XP classloader

### ISSUE THREAD

**Issue by thekid** [xp-forge/mirrors#13]
* [x] Native classes (via reflection)
* [x] Foreign classes

**Comment by thekid** [comment]
```sh
Timm@slate ~/devel/stubbles-core [master]
$ xp -cp ../xp/parse/src/main/php/ -cp ../xp/mirrors/src/main/php/ -w '
  require("vendor/autoload.php");
  return (new \lang\mirrors\TypeMirror("stubbles\\lang\\Sequence"))->methods()->named("of")
'
lang.mirrors.Method(public static stubbles.lang.Sequence of(stubbles.lang.Sequence|.Traversable|array $elements))
```

OK (except that type unions are ot supported).

**Comment by thekid** [comment]
For type unions to work, we'd need https://github.com/xp-framework/core/pull/76

**Comment by thekid** [comment]
:white_check_mark: Type unions are now supported

Other common things in the PHP world are types like "mixed" ("var" in XP speak) and "null" (not really a type, could be translated to "var" I guess), see e.g. http://www.phpdoc.org/docs/latest/guides/types.html#keywords

**Comment by thekid** [comment]
Fixed a bug with fields' types not correctly being resolved in xp-forge/sequence@8f0f49e

**Comment by thekid** [comment]
:white_check_mark: Now supports everything http://www.phpdoc.org/ lists under its type section. The other alternative implementations:

* http://www.phpdoc.de/ looks unmaintained
* https://github.com/docblox/docblox - merged to phpDocumentor2 

...so it seems that the types list is authoritative for PHP. With the two fixes in xp-forge/sequence@76e4afe (for the `var` tag) and xp-forge/sequence@13e254a (for whitespace) I think we can claim completion.
